### DISCUSSION TITLE
Constraints on the boundary

### DISCUSSION THREAD

**Discussion by alessiofumagalli** [pmgbergen/porepy#612] [Q&A]
Dear all, is it possible to put constraints on the boundary of a 3d grid? Thanks in advance for your help

**Comment by jhabriel** [comment]
Hi @alessiofumagalli. As far as I know, this is not possible. I've tried my self to impose these constraints in specific parts of a 3D boundary in the past (for the 3D Geiger benchmark problem to be specific), but PorePy threw an error related to mesh generation. I'm sure @keileg will have further inputs on this.

**Reply by alessiofumagalli** [reply]
Thanks a lot @jhabriel for your quick reply, I got the same experience but I need to impose some constraints on a portion of a boundary for one of my current projects. Maybe @keileg has some suggestions :)

**Comment by keileg** [comment]
I am afraid @jhabriel is correct that this is not straightforward. The fracture intersection algorithm will crash spectacularly if you feed it planes that are overlapping; this is a whole new level of complexity that I have promissed myself never to look into.

If what you really want is to make the grid on the boundary conform to certain constraints, I have two suggestions for workounds, though:
1. Split the boundary surface into subsurfaces, so that the subsurface boundaries form the constraints you want. This should work, but may be a bit tedious.
2. Add constraints to the fracture network that cross the boundary at the target lines. These auxiliary constraints need not extend far into the domain (though you will get small cells if you make a small constraint), so with some luck the constraints may not cross other fractures (that could potentially lead to artificially difficult meshing on the real fractures).

I think both of these should work, but I can't promise that you will not hit some additional difficulties, e.g., the boundary and/or fracture description get more involved. It is hard to tell how much of a problem this will be.

Let me know if you try any of the above, and if so, how it goes. There is a longstanding issue in another repo (https://github.com/keileg/porepy_mesh_factory/issues/1) that could be solved with such a hack if it works.
